aiohttp/cookiejar.py

```diff
@@ -48,15 +47,16 @@ class CookieJar(AbstractCookieJar):
 
     DATE_YEAR_RE = re.compile(r"(\d{2,4})")
 
-    MAX_TIME = 2051215261.0  # so far in future (2035-01-01)
+    MAX_TIME = datetime.datetime.max.replace(
+        tzinfo=datetime.timezone.utc)
 
     def __init__(self, *, unsafe: bool=False) -> None:
         self._loop = get_running_loop()
         self._cookies = defaultdict(SimpleCookie)  #type: DefaultDict[str, SimpleCookie]  # noqa
         self._host_only_cookies = set()  # type: Set[Tuple[str, str]]
         self._unsafe = unsafe
-        self._next_expiration = ceil(self._loop.time())
-        self._expirations = {}  # type: Dict[Tuple[str, str], int]
+        self._next_expiration = self._next_whole_second()
+        self._expirations = {}  # type: Dict[Tuple[str, str], datetime.datetime]
```

Comment:
This line is now too long for flake8. Line 56 had the same issue and used a noqa flag to tell flake8 to ignore it. We could do the same here but I’d prefer it to be specific:

```suggestion
        self._expirations = {}  # type: Dict[Tuple[str, str], datetime.datetime] # noqa: E501
```

@asvetlov remind me to talk to you about flake8-bugbear and the flexible line length overflow setting. :-) if this project is to use black in future that’s a good addition anyway.

Reply:
Change made before I saw the suggestion. It's also the reason I went to inline type hints. ;)